### ISSUE TITLE
[v4] Arrow functions won't revert back to original state

### ISSUE THREAD

**Issue by Kimahriman** [gaearon/react-hot-loader#836]
### Description

This is a weird bug to describe, but if you modify an arrow function and then change it back to it's original, the second update won't work. I can try to put up a full working repo if it's not clear from my example.

### Example

Hello.js:
```typescript
import React from 'react'

class Hello extends React.Component {

  getText = () => {
    return "Hello world"
  }

  render() {
    return this.getText()
  }
}

export default Hello
```

### Expected behavior

Changing the text "Hello world" to "Hello RHL" will change the text in the browser to "Hello RHL". Then changing the text back to "Hello world" will change the text in the browser to "Hello world".

### Actual behavior

Changing the text "Hello world" to "Hello RHL" changes the text in the browser to "Hello RHL". Then changing the text back to "Hello world" causes no change in the browser, even though the console indicates that the correct modules were updated.

This isn't just a problem with multiple updates. If you change the text to something new twice, and then back to "Hello world", the first two updates will work, but not the final one.

### Environment

React Hot Loader version: 4.0.0-beta.18

Run these commands in the project folder and fill in their results:

1. `node -v`: v8.4.0
2. `npm -v`: 5.6.0

Then, specify:

1. Operating system: OSX Sierra
2. Browser and version: 63.0.3239.132

**Comment by theKashey** [comment]
Confirmed. Logic this not detect any difference from the original class, ignoring injected into instance variant.

**Comment by neoziro** [comment]
Fixed in v4.0.0-beta.19.
